src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -85,15 +90,16 @@ public UserDto updateUserInfo(@Valid @RequestBody UserDto userDto, Principal pri
     @ApiOperation(value = "Add children data for a registered requested user")
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 400, message = "BAD REQUEST")})
+            @ApiResponse(code = 400, message = "BAD REQUEST"),
+            @ApiResponse(code = 406, message = "NOT_ACCEPTABLE")})
     @ResponseStatus(HttpStatus.OK)
-    public void addChildren(@RequestBody @Size(min = 1, message = "{exception.invalid.dto}")
-                                        List<@Valid ChildDto> childrenDto, Principal principal) {
-        if (childrenDto.stream().anyMatch(dto -> dto.getId() > 0)) {
-            throw new IllegalIdentifierException(
-                    getExceptionMessageSourceWithAdditionalInfo(
-                            "exception.illegal.id",
-                            "Zero ID is expected"));
+    @Validated({Default.class, New.class})
+    public void addChildren(@RequestBody @Size(min = 1, max = MAX_CHILDREN_AMOUNT, message = "{exception.invalid.dto}")
+                                 List<@Valid ChildDto> childrenDto, Principal principal) throws EntityAmountException {
+        int amountOfChildren = childrenDto.size() + userService.getChildren(getUser(principal.getName())).size();
```

Comment:
You just need to get exiting children amount from existed user. You don't need extra type convention for that case. Make it simpler: `getUser(principal.getName()).getChildren().size();`

Reply:
Nice catch! I will fix this